go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -384,15 +388,17 @@ func (s *VtctldServer) ApplyVSchema(ctx context.Context, req *vtctldatapb.ApplyV
 			if len(ups) == 0 {
 				continue
 			}
-			unknownVindexParams[name] = &vtctldatapb.ApplyVSchemaResponse_ParamList{Params: ups}
+			response.UnknownVindexParams[name] = &vtctldatapb.ApplyVSchemaResponse_ParamList{Params: ups}
+			unknownVindexParams = append(unknownVindexParams, fmt.Sprintf("%s (%s)", name, strings.Join(ups, ", ")))
 		}
 	}
 
-	if req.DryRun { // we return what was passed in and parsed, plus unknown vindex params
-		return &vtctldatapb.ApplyVSchemaResponse{
-			VSchema:             vs,
-			UnknownVindexParams: unknownVindexParams,
-		}, nil
+	if req.DryRun || (req.Strict && len(unknownVindexParams) > 0) { // return early if dry run or unknown params in strict mode
+		var err error
+		if req.Strict && len(unknownVindexParams) > 0 {
+			err = fmt.Errorf("unknown vindex params: %s", strings.Join(unknownVindexParams, "; "))
```

Comment:
I think it would be better to use a vterror here with the INVALID_ARGUMENT code.